src/countries/countries.controller.ts

```diff
@@ -1,7 +1,50 @@
 import { CountriesService } from './countries.service';
-import { Controller } from '@nestjs/common';
+import { Body, Controller, Delete, Get, Param, Post, Put, Req } from "@nestjs/common";
+import { CreateCountryDto } from './dto/create-country.dto';
+import { CountryInterface } from './country.interface';
+import { UpdateCountryDto } from './dto/update-country.dto';
+import { DeleteResult, UpdateResult } from 'typeorm';
 
-@Controller()
+@Controller('countries')
 export class CountriesController {
   constructor(private readonly countryService: CountriesService) {}
+
+  @Get()
+  async findAll() {
+    const countries: Array<CountryInterface> =
+      await this.countryService.findAll();
+    return countries;
+  }
+
+  @Get(':id')
```

Comment:
This might be a personal preference, but I believe it's a better coding standard to have a proper errors on failure, rather than returning null.

If I'm to rewrite this function I'll write it like this:
```typescript
@Get(':id')
  async find(@Param('id') id: string): Promise<CountryDto> {
    try {
      return await this.countryService.findOne(+id);
    } catch (error) {
      if (error instanceof CountryNotFoundError) {
        throw new NotFoundException(error.message);
      }
      throw new InternalServerErrorException();
    }
  }
```

Where the service function is refactored to be like this:
```typescript
  async findOne(id: number): Promise<CountryDto> {
    const country = await this.countryRepository.findOneBy({ id });
    if (!country) throw new CountryNotFoundError(id);
    return country;
  }
```

And the `CountryNotFoundError` is defined like this:
```typescript
// src/countries/errors/country-not-found.error.ts
export class CountryNotFoundError extends Error {
  constructor(countryId: string) {
    super(`Country with id ${countryId} not found`);
  }
}
```

And imported like this:
```typescript
import { CountryNotFoundError } from './errors';
```

Again, the reason we have `'./errors'` is because inside the errors folder we exported everything within the index file:
```typescript
// src/countries/errors/index.ts
export * from './country-not-found.error';
```